File: Fieldium/Model/Entities/Voucher.swift
//
//  Voucher.swift
//  Fieldium
//
//  Created by Yahya Tabba on 6/11/17.
//  Copyright © 2017 Yahya Tabba. All rights reserved.
//

import Foundation
import ObjectMapper
import SwiftyJSON

class Voucher : BaseEntity {
    
    var all_games : JSON!
    var voucher_id : JSON!
    var type : JSON!
    var value : JSON!
    var valid : JSON!
    
    var voucher : String!
    var creation_date : String!
    var from_hour : String!
    var to_hour : String!
    var start_date : String!
    var expiry_date : String!

    var description_en : String!
    var description_ar : String!

    var companies = [Company]()
    var games = [Game]()
    var message : String!

    
    // Mappable
    override func mapping(map: Map) {
        all_games <- map["all_games"]
        voucher_id <- map["voucher_id"]
        type <- map["type"]
        value <- map["value"]
        valid <- map["valid"]
        
        voucher <- map["voucher"]
        creation_date <- map["creation_date"]
        from_hour <- map["from_hour"]
        to_hour <- map["to_hour"]
        start_date <- map["start_date"]
        expiry_date <- map["expiry_date"]
        
        description_en <- map["description_en"]
        description_ar <- map["description_ar"]
        
        companies <- map["companies"]
        games <- map["games"]
        
        message <- map["message"]
    }
    
}

File: app/src/main/java/com/bluebrand/fieldium/core/model/Voucher.java
package com.bluebrand.fieldium.core.model;

import android.view.View;

import java.io.Serializable;
import java.math.BigDecimal;
import java.util.ArrayList;

/**
 * Created by Player on 12/22/2016.
 */
public class Voucher implements Serializable {
    private int id;
    private int valid;
    private int type;
    private String code;
    private String from_hour;
    private String to_hour;
    private String start_date;
    private String expiry_date;
    private int value;
    private String description;
    private ArrayList<Company> companies;
    private ArrayList<Game> games;
    private String message;
    private View.OnClickListener requestBtnClickListener;

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }

    public int getValid() {
        return valid;
    }

    public void setValid(int valid) {
        this.valid = valid;
    }

    public int getType() {
        return type;
    }

    public void setType(int type) {
        this.type = type;
    }

    public int getValue() {
        return value;
    }

    public void setValue(int value) {
        this.value = value;
    }

    public String getExpiry_date() {
        return expiry_date;
    }

    public void setExpiry_date(String expiry_date) {
        this.expiry_date = expiry_date;
    }

    public String getStart_date() {
        return start_date;
    }

    public void setStart_date(String start_date) {
        this.start_date = start_date;
    }

    public String getCode() {
        return code;
    }

    public void setCode(String code) {
        this.code = code;
    }

    public String getFrom_hour() {
        return from_hour;
    }

    public void setFrom_hour(String from_hour) {
        this.from_hour = from_hour;
    }

    public String getTo_hour() {
        return to_hour;
    }

    public void setTo_hour(String to_hour) {
        this.to_hour = to_hour;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getImageUrl() {
        return from_hour;
    }

    public void setImageUrl(String imageUrl) {
        this.from_hour = imageUrl;
    }

    public ArrayList<Company> getCompanies() {
        return companies;
    }

    public void setCompanies(ArrayList<Company> companies) {
        this.companies = companies;
    }


    public View.OnClickListener getRequestBtnClickListener() {
        return requestBtnClickListener;
    }

    public void setRequestBtnClickListener(View.OnClickListener requestBtnClickListener) {
        this.requestBtnClickListener = requestBtnClickListener;
    }

    public Voucher() {
        companies = new ArrayList<>();
        games = new ArrayList<>();
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public ArrayList<Game> getGames() {
        return games;
    }

    public void setGames(ArrayList<Game> games) {
        this.games = games;
    }
}
